test/sql/collate/test_collate_orderby_column_number.test

```diff
@@ -0,0 +1,38 @@
+# name: test/sql/collate/test_collate_orderby_column_number.test
+# description: Test collation on ORDER BY with column number
+# group: [collate]
+
+statement ok
+PRAGMA enable_verification
+
+query I
+SELECT 'a' AS c1 ORDER BY 1 COLLATE NOCASE;
+----
+a
+
+query I
+SELECT 'a' ORDER BY 1 COLLATE NOCASE;
+----
+a
+
+statement ok
+CREATE TABLE collate_test(s VARCHAR)
+
+statement ok
+INSERT INTO collate_test VALUES ('ã'),('B'),('a'),('A');
+
+query I
+SELECT s FROM collate_test ORDER BY 1 COLLATE NOCASE;
```

Comment:
In addition to the tests @Tishj requested could we add some tests with more complex expressions as well? i.e.

```sql
SELECT CONCAT(s, s) FROM collate_test ORDER BY 1 COLLATE NOCASE;
SELECT (SELECT s) FROM collate_test ORDER BY 1 COLLATE NOCASE;
SELECT CONCAT(s, s) AS concat, concat FROM collate_test ORDER BY 2 COLLATE NOCASE;
SELECT CASE WHEN s[1]='a' THEN s ELSE NULL END FROM collate_test ORDER BY 1 COLLATE NOCASE;
SELECT collate_test.s FROM collate_test ORDER BY 1 COLLATE NOCASE;
```